### PR TITLE
[RAINCATCH-1101] Move passport integration to an auth module

### PR DESCRIPTION
**Pull request by JameelB**
## Motivation
Passport will need to be integrated with both the mobile and portal app. An auth module which can be consumed by both the mobile and portal demo application needs to be created to reduce code duplication.

## Description
Moved the current Passport integration in the mobile app to a separate auth module which the mobile and portal can use.

## Progress
- [ ] Create a separate demo auth module
- [ ] Move getProfile and logout implementation to the auth module
- [ ] Integrate the demo auth module with the mobile app

## Additional Notes
Related JIRA Ticket - https://issues.jboss.org/browse/RAINCATCH-1101

### CODE REVIEW

File: demo/auth/authService.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+var fh = require('fh-js-sdk');
+
+function AuthService() {
+}
+
+AuthService.prototype.getProfile = function($http, $window) {
+  var req = {
+    method: 'GET',
+    url: fh.getCloudURL()+ '/profile'
```
Comment (wtrocki) [inline demo/auth/authService.js]: That's wrong usage of the fh api. We should wait for init and then execute any methods. 
I'm surprised that works.

File: demo/auth/authService.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  };
+  return $http(req, {withCredentials: true}).then(function(res) {
```
Comment (wtrocki) [inline demo/auth/authService.js]: Format this file

File: demo/auth/authService.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return res.data;
+  }, function(err) {
+    if (err.status === 401) {
+      $window.location = fh.getCloudURL() + '/login';
+    }
+    if (err.status === 403) {
+      console.log('Forbidden')
```
Comment (wtrocki) [inline demo/auth/authService.js]: Any forbidden call will be visible in console anyway, but none of the applications users sees console. Why not use logger? IMHO Cordova "event dialog" will be perfect for this.

File: demo/auth/authService.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+var fh = require('fh-js-sdk');
```
Comment (wtrocki) [inline demo/auth/authService.js]: We may move this and wfm module to `components` folder. It feel strange that we have it here and I missed that when reviewing

File: demo/auth/authService.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+AuthService.prototype.logout = function($http, $window) {
+  var req = {
+    method: 'GET',
+    url: fh.getCloudURL() + '/logout'
+  };
+
+  return $http(req, {withCredentials: true}).then(function(res) {
+    $window.location = fh.getCloudURL() + '/login';
+  }, function(err) {
+    console.log('error logging out');
```
Comment (wtrocki) [inline demo/auth/authService.js]: Let's use logger (btw. console.error is something you should use for logging errors)

File: demo/auth/authService.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  });
+};
+
+angular.module('wfm.common.apiservices').service('authService', function() {
```
Comment (wtrocki) [inline demo/auth/authService.js]: When moving to separate module it's not part of the `common.apiservices` module. Especially that you creating separate module bellow.

**Review by wtrocki — CHANGES_REQUESTED** [review]

**Comment by JameelB** [comment]
Closed for https://github.com/TommyJ1994/raincatcher-angularjs/pull/2

**Pull request closed without merging**